esp32/mphalport.c

```diff
@@ -110,10 +119,13 @@ void mp_hal_delay_ms(uint32_t ms) {
         // do the remaining delay accurately
         ets_delay_us((ms - dt) * 1000);
     }
+    MP_THREAD_GIL_ENTER();
 }
 
 void mp_hal_delay_us(uint32_t us) {
+    MP_THREAD_GIL_EXIT();
```

Comment:
For efficiency and accuracy I would not release the GIL during a microsecond sleep.